felix/dataplane/linux/qos/qos.go

```diff
@@ -315,13 +350,19 @@ func UpdateEgressQdisc(tbs *TokenBucketState, ifbDeviceName string) error {
 
 func GetTBFValues(rateBitsPerSec, burstBits uint64) *TokenBucketState {
 	rateBytesPerSec := rateBitsPerSec / 8
+
 	burstBytes := burstBits / 8
 
-	time2Tick := uint32(float64(burstBytes) * float64(1000000) / float64(rateBytesPerSec))
-	bufferBytes := uint32(float64(time2Tick) * float64(netlink.TickInUsec()))
+	// Time (in usec) it takes to transmit the burst size at the given rate
+	timeToBurstUsec := uint32(float64(burstBytes) * float64(1000000) / float64(rateBytesPerSec))
+
+	// Buffer size needed, obtained by multiplying timeToBurstUsec by the number of usec per tick of the network scheduler
```

Comment:
Thanks.  I understand now that this is the number of ticks needed to transmit (or perhaps to receive) a burst at the specified rate.  But why does that give us a buffer size?

Reply:
isn't that the number of bytes that can be transmitted per per tick of the network scheduler (at the configured burst and rate above)? so that would be the necessary buffer size to accumulate data in-between ticks?

Reply:
I still don't feel like I understand this, but that's not a reason to hold up the PR, and it sounds like you _do_ understand!